lib/core/diagnostics.js

```diff
@@ -23,7 +23,13 @@ const channels = {
   close: diagnosticsChannel.channel('undici:websocket:close'),
   socketError: diagnosticsChannel.channel('undici:websocket:socket_error'),
   ping: diagnosticsChannel.channel('undici:websocket:ping'),
-  pong: diagnosticsChannel.channel('undici:websocket:pong')
+  pong: diagnosticsChannel.channel('undici:websocket:pong'),
+  // Fetch channels
+  start: diagnosticsChannel.channel('undici:fetch:start'),
+  end: diagnosticsChannel.channel('undici:fetch:end'),
+  asyncStart: diagnosticsChannel.channel('undici:fetch:asyncStart'),
+  asyncEnd: diagnosticsChannel.channel('undici:fetch:asyncEnd'),
+  fetchError: diagnosticsChannel.channel('undici:fetch:fetchError')
```

Comment:
does the channel really need to be named 'undici:fetch:fetchError'?
I would prefer

'undici:fetch:error'

as it is clear that we are in fetch

```suggestion
  fetchError: diagnosticsChannel.channel('undici:fetch:error')
```

Reply:
It most certainly does not, this was just for consistency. I have changed it just error 👍🏼